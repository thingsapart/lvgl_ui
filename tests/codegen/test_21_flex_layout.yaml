# A container with a flex layout (row) and children with different `flex_grow` properties.
- type: obj
  size: [300, 80]
  center: []
  layout: LV_LAYOUT_FLEX
  flex_flow: LV_FLEX_FLOW_ROW
  pad_all: 5
  pad_gap: 5
  children:
    - type: button
      flex_grow: 1 # Takes 1 part of the space
      children:
        - type: label
          text: '1'
          center: []
    - type: button
      flex_grow: 2 # Takes 2 parts of the space
      children:
        - type: label
          text: '2'
          center: []
    - type: button
      flex_grow: 1 # Takes 1 part of the space
      children:
        - type: label
          text: '1'
          center: []
