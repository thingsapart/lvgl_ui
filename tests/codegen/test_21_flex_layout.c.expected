/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: obj_0 (obj)
    lv_obj_t* obj_0 = lv_obj_create(parent);

    lv_obj_set_style_size(obj_0, 300, 80, 0);
    lv_obj_center(obj_0);
    lv_obj_set_style_layout(obj_0, LV_LAYOUT_FLEX, 0);
    lv_obj_set_style_flex_flow(obj_0, LV_FLEX_FLOW_ROW, 0);
    lv_obj_set_style_pad_all(obj_0, 5, 0);
    lv_obj_set_style_pad_gap(obj_0, 5, 0);
    // unnamed: button_1 (button)
    lv_obj_t* button_1 = lv_button_create(obj_0);

    lv_obj_set_style_flex_grow(button_1, 1, 0);
    // unnamed: label_2 (label)
    lv_obj_t* label_2 = lv_label_create(button_1);

    lv_label_set_text(label_2, "1");
    lv_obj_center(label_2);


    // unnamed: button_3 (button)
    lv_obj_t* button_3 = lv_button_create(obj_0);

    lv_obj_set_style_flex_grow(button_3, 2, 0);
    // unnamed: label_4 (label)
    lv_obj_t* label_4 = lv_label_create(button_3);

    lv_label_set_text(label_4, "2");
    lv_obj_center(label_4);


    // unnamed: button_5 (button)
    lv_obj_t* button_5 = lv_button_create(obj_0);

    lv_obj_set_style_flex_grow(button_5, 1, 0);
    // unnamed: label_6 (label)
    lv_obj_t* label_6 = lv_label_create(button_5);

    lv_label_set_text(label_6, "1");
    lv_obj_center(label_6);



}
